crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -86,16 +86,16 @@ pub struct Immutable {
 }
 
 /// A shared object is an object that is shared using the 0x2::transfer::share_object function
-/// and is accessible to everyone.
-/// Unlike owned objects, once an object is shared, it stays mutable and can be accessed by anyone,
-/// unless it is made immutable. An example of immutable shared objects are all published packages
-/// and modules on Sui.
+/// and is accessible to everyone. There are two types of shared objects: mutable and immutable.
+/// Unlike owned objects, once an object is shared, it stays mutable and can be accessed by anyone.
+/// An example of immutable shared objects are all published packages and modules on Sui.
```

Comment:
It's confusing to refer to "immutable" objects here, because they would use the `Immutable` owner type (A published package's `owner.__typename` would be `Immutable`).  Let's just keep this section about shared objects, we don't really conceptualise immutable as a kind of shared object.

The reason why we try and distance the two concepts is because while they are both readable by all transactions:

- Shared objects can be modified.
- Shared objects must go through consensus, even when only read, while Immutable objects can be used in fast path transactions.

The second bullet is probably the most crucial difference between them (without this detail, we couldn't use Move code in the single owner fast path, which would be a pretty dire compromise).

```suggestion
/// and is accessible to everyone.
/// Unlike owned objects, once an object is shared, it stays mutable and can be accessed by anyone.
```

(+ line wrapping changes).

Reply:
> Shared objects must go through consensus, even when only read, while Immutable objects can be used in fast path transactions.

This is a very important distinction, thank you.